src/openrct2/scripting/ScTile.hpp

```diff
@@ -719,37 +910,113 @@ namespace OpenRCT2::Scripting
                     Invalidate();
                     break;
                 }
+                case TILE_ELEMENT_TYPE_WALL:
+                {
+                    auto el = _element->AsWall();
+                    el->SetSecondaryColour(value);
+                    Invalidate();
+                    break;
+                }
+            }
+        }
+
+        uint8_t tertiaryColour_get() const
+        {
+            auto el = _element->AsWall();
+            if (el != nullptr)
+                return el->GetTertiaryColour();
+            return 0;
+        }
+        void tertiaryColour_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto el = _element->AsWall();
+            if (el != nullptr)
+            {
+                el->SetTertiaryColour(value);
+                Invalidate();
+            }
+        }
+
+        uint16_t bannerIndex_get() const
+        {
+            switch (_element->GetType())
+            {
+                case TILE_ELEMENT_TYPE_LARGE_SCENERY:
+                {
+                    auto el = _element->AsLargeScenery();
+                    return el->GetBannerIndex();
+                }
+                case TILE_ELEMENT_TYPE_WALL:
+                {
+                    auto el = _element->AsWall();
+                    return el->GetBannerIndex();
+                }
+                case TILE_ELEMENT_TYPE_BANNER:
+                {
+                    auto el = _element->AsBanner();
+                    return el->GetIndex();
+                }
+            }
+            return 0;
+        }
+        void bannerIndex_set(uint16_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            switch (_element->GetType())
+            {
+                case TILE_ELEMENT_TYPE_LARGE_SCENERY:
+                {
+                    auto el = _element->AsLargeScenery();
+                    el->SetBannerIndex(value);
+                    Invalidate();
+                    break;
+                }
+                case TILE_ELEMENT_TYPE_WALL:
+                {
+                    auto el = _element->AsWall();
+                    el->SetBannerIndex(value);
+                    Invalidate();
+                    break;
+                }
+                case TILE_ELEMENT_TYPE_BANNER:
+                {
+                    auto el = _element->AsBanner();
+                    el->SetIndex(value);
+                    Invalidate();
+                    break;
+                }
             }
         }
 
-        bool railings_get() const
+        uint8_t edges_get() const
         {
             auto el = _element->AsPath();
-            return el != nullptr ? el->GetRailingEntryIndex() : false;
+            return el != nullptr ? el->GetEdges() : 0;
         }
-        void railings_set(bool value)
```

Comment:
Why did you remove railings_get / railings_set ?

Reply:
It's unused. The getter is just an alias for `GetSurfaceEntryIndex` and the setter does nothing.

Reply:
.. for now. But well, how path surfaces and railings work is bound to change a lot anyway.